src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1792,20 +1829,112 @@ private void setItemToCacheFromRefresh(final RefreshItem refreshItem,
             logReturnedToken(request, result);
 
             // Update for cache key
+            // If token is refreshed with Family token, make sure the tokenCacheItem doesn't
+            // contain the client id. 
+            final TokenCacheItem tokenCacheItem = new TokenCacheItem(request, result, refreshItem.mMultiResource);
+            if (tokenRefreshedWithFRT(request, refreshItem)) {
+                tokenCacheItem.setClientId(null);
```

Comment:
that doesn't look well. this property should be set to correct one in TokenCacheItem constructor